app/backend/approaches/chatreadretrieveread.py

```diff
@@ -108,7 +111,14 @@ def run(self, history: list[dict], overrides: dict) -> any:
     def get_chat_history_as_text(self, history, include_last_turn=True, approx_max_tokens=1000) -> str:
         history_text = ""
         for h in reversed(history if include_last_turn else history[:-1]):
-            history_text = """<|im_start|>user""" +"\n" + h["user"] + "\n" + """<|im_end|>""" + "\n" + """<|im_start|>assistant""" + "\n" + (h.get("bot") + """<|im_end|>""" if h.get("bot") else "") + "\n" + history_text
-            if len(history_text) > approx_max_tokens*4:
-                break    
+            history_text = f"""\
+<|im_start|>user
+{h["user"]}
+<|im_end|>
+<|im_start|>assistant
+{h.get("bot") + "<|im_end|>" if h.get("bot") else ""}
```

Comment:
curious - is this the intended behaviour?

1. Shouldn't we accumulate chat history and do `history_text +=`?
2. Should `<|im_start|>assistant` be left open if `h.get("bot")` evaluates to False?